src/big_int.rs

```diff
@@ -0,0 +1,764 @@
+use crate::CrateError;
+use gmp::{
+    mpz::{Mpz, ProbabPrimeResult},
+    rand::RandState,
+};
+use num::traits::{One, Zero};
+use std::{
+    collections::hash_map::DefaultHasher,
+    hash::{Hash, Hasher},
+    iter::{Product, Sum},
+    ops::{Deref, Mul, MulAssign, Neg},
+};
+
+#[derive(Debug, Clone)]
+pub struct RNSRepresentation {
```

Comment:
this file is pretty big, do you think it could be interesting to move `RNSRepresentation` in its own module or not?

Reply:
Why not